### DISCUSSION TITLE
Dev mode server and websockets

### DISCUSSION THREAD

**Discussion by mbjeski** [vercel/next.js#33844] [Help]
I've implemented a proxy to my back-end GraphQL API inside a NextJS API (/api/graphqlws) using http-proxy-middleware. The application works as expected when I run in non-development mode (npm run build, npm run start). But, when I run in development mode with HMR, the websocket client cannot connect to the server. It seems like the NextJS dev server is handling the websocket request and closing it before it can reach my configured api proxy. Is there a way to get this to work ?

**Comment by utiq** [comment]
@mbjeski did you solve the issue? I'm having the same problem
